rfcs/proposed/task_group_dynamic_dependencies/implementation_details.md

```diff
@@ -0,0 +1,516 @@
+The submission function must check whether the owned task has any dependencies (i.e., whether `m_num_dependencies` is non-zero) and, if so,
+decrement the counter.
+
+If the task's dynamic state is not initialized, it cannot have any dependencies and is spawned immediately.
+
+If the dependency counter reaches zero, the task is ready to be spawned. Otherwise, it will be spawned or bypassed by the last predecessor that releases its list
+of successors.
+
+## Transferring completion to the other task
+
+The API `tbb::task_group::transfer_this_task_completion_to(tbb::task_handle& new_task)` operates by setting `m_new_dynamic_state` in
+the dynamic state of the currently executing task to point to the dynamic state of `new_task`.
+
+To determine the dynamic state associated with the currently executing task, a pointer to that task must be retrieved from the task scheduler.
+This is accomplished via the [new entry point](#library-abi-changes) `tbb::task* current_task_ptr()` is added. 
+
+The `transfer_this_task_completion_to` transfers the successors from the currently executing task to `new_task`.
+If the dynamic state of the currently executing task has not been initialized, it cannot have any associated successors.
+
+During the transfer, the `m_new_dynamic_state` pointer is set to the dynamic state of `new_task`. The successors
+are then retrieved from the original successor list and appended to the successor list of  `new_dynamic_state`.
+
+The head pointer of the successor list is exchanged to expose the `transferred` state.
+
+Successor list insertion into `new_task`'s dynamic state follows the same logic as `set_task_order`, but skips additional checks, as the recipient task
+is guaranteed to be in the `created` state and it's state cannot change.
+
+The sequence diagram for `transfer_this_task_completion_to` is shown below:
+
+<img src="assets/impl_sequence_diagram_transfer.png">
+
+### Dynamic state lifetime issue
+
+During the implementation of `transfer_this_task_completion_to`, a lifetime management issue was identified in `task_dynamic_state`.
+
+Initially, its reference counter was incremented only when the task or a `task_completion_handle` was created.
+
+Consider the following example: task `A` and a `task_completion_handle` named `A_comp_handle`, initially assigned to handle the completion of `A`. 
+The dynamic state `A_state` is associated with task `A`. 
+
+While task `A` is executing, it transfers its completion to task `B` (illustrated by the red arrow in the diagram below).
+The dynamic state `B_state` is associated with task `B`.
+
+As described earlier, `A_state::m_new_dynamic_state` is set to point to `B_state` (as shown by the green arrow in the diagram).
+
+<img src="assets/impl_lifetime_issue.png" width=800>
+
+After task `A` completes, task `B` is executed and subsequently destroyed, along with its associated `B_state`.
```

Comment:
I would add that it is destroyed because there are no completion handles for this task (and we know that it's the typical case for continuations).

Reply:
Updated.